src/data/flamegraphs.rs

```diff
@@ -31,39 +40,67 @@ impl FlamegraphRaw {
 }
 
 impl CollectData for FlamegraphRaw {
+    fn prepare_data_collector(&mut self, _params: CollectorParams) -> Result<()> {
+        match Command::new("perf").args(["--version"]).output() {
+            Err(e) => Err(PDError::DependencyError(format!("'perf' command failed. {}", e)).into()),
+            _ => Ok(()),
+        }
+    }
+
     fn after_data_collection(&mut self, params: CollectorParams) -> Result<()> {
-        let data_dir = PathBuf::from(params.data_dir.clone());
+        let data_dir = PathBuf::from(&params.data_dir);
 
-        let mut file_pathbuf = data_dir.clone();
-        file_pathbuf.push(get_file_name(
-            params.data_dir.clone(),
-            "perf_profile".to_string(),
-        )?);
+        let file_pathbuf =
+            data_dir.join(get_file_name(params.data_dir, "perf_profile".to_string())?);
 
-        let mut perf_jit_loc = data_dir.clone();
-        perf_jit_loc.push(format!("{}-perf.data.jit", params.run_name));
+        let perf_jit_loc = data_dir.join("perf.data.jit");
 
-        println!("Running Perf inject...");
+        trace!("Running Perf inject...");
         let out_jit = Command::new("perf")
             .args([
                 "inject",
                 "-j",
                 "-i",
                 &file_pathbuf.to_str().unwrap(),
                 "-o",
-                perf_jit_loc.clone().to_str().unwrap(),
+                perf_jit_loc.to_str().unwrap(),
             ])
-            .status();
+            .output();
+
+        let fg_out = File::create(data_dir.join(format!("{}-flamegraph.svg", params.run_name)))?;
+
         match out_jit {
             Err(e) => {
-                if e.kind() == ErrorKind::NotFound {
-                    error!("'perf' command not found.");
-                } else {
-                    error!("Unknown error: {}", e);
+                let out = format!("Skip processing profiling data due to: {}", e);
+                error!("{}", out);
+                write_msg_to_svg(fg_out, out)?;
+            }
+            Ok(_) => {
+                info!("Creating flamegraph...");
+                let script_loc = data_dir.join("script.out");
+                let out = Command::new("perf")
+                    .stdout(File::create(&script_loc)?)
+                    .args(["script", "-f", "-i", perf_jit_loc.to_str().unwrap()])
```

Comment:
For the other invocations of perf you use it's -o option rather than stdout. I don't think there's a functional problem here, but it would be better to be consistent on how you're calling it.

Reply:
Most other perf sub-tools provide the -o option. perf script does not :(